### PR TITLE
Move modified HpaMaxedOut alert to kube-prometheus-stack helm chart

### LINKED ISSUE #575: KubeHpaMaxedOut patching breaks any alterations via user-values file.
Currently the default set of alerts is deployed using helm to the cluster and then patched inside the cluster to fix an issue with the Kube HPA alert using `kubectl patch`. This patch specifically references the 14th element in the resulting PrometheusRule manifest. This makes it impossible to modify the list of deployed alerts using parameters to the kube-prometheus-stack via the user-values file.

https://github.com/sassoftware/viya4-monitoring-kubernetes/blob/215be0ce522b354175fea58791180885ee335ad6/monitoring/kube-hpa-alert-patch.json#L4

A better approach is disabling the deployment of the KubeHpaMaxedOut alert via [the defaultRules.disabled variable](https://github.com/prometheus-community/helm-charts/blob/1ff5e72a287484ae817c234ab2623cb2b2f91eb5/charts/kube-prometheus-stack/values.yaml#L151) and adding a modified version of it using [the appropriate option](https://github.com/prometheus-community/helm-charts/blob/1ff5e72a287484ae817c234ab2623cb2b2f91eb5/charts/kube-prometheus-stack/values.yaml#L167).

Using this approach is preferred as it also prevents deploying incorrect rules to the cluster intermittently.

### PR DESCRIPTION
**Pull request by tynsh**
Currently the HpaMaxedOut alert from the default rules set of the kube-prometheus-stack helm chart is deployed and then modified on the cluster. More information on the current situation can be found in PR #575.

This PR moves the change to the values-prom-operator.yaml file so that users can modify the default set of alerts. The Alert is renamed to reflect the modification and avoid issues with the reused name during upgrades.

Fixes #575.

### CODE REVIEW

**Comment by gsmith-sas** [comment]
@tynsh Thank you for submitting this PR to address the problem you reported in #575.  We will review the changes and do some testing/validation before merging it into the project.  It is a short work here in the US due to a two-day holiday so we probably won't be able to do that until next week at the earliest.  Thank you once again for taking the time to craft a fix and submitting it.
Regards,
Greg

**Comment by tynsh** [comment]
@gsmith-sas No problem. Take your time :)
Best,
Tobias

**Review by gsmith-sas — APPROVED** [review]

**Comment by gsmith-sas** [comment]
We have approved this PR and merged the changes into our 'main' branch after validating it works as intended.  

@tynsh Thank you for your contributions and (once again) making our project better!

**Comment by gsmith-sas** [comment]
@tynsh I just noticed that this Pull Request did not include the required contributor sign-off text.  Unfortunately, I noticed this _after_ merging the PR into the 'main' branch, so I had to revert (back out) the changes.  Please open a new Pull Request with these same changes and include the contributor sign-off text.  As a reminder, the contributor sign-off requires adding text like the following to the DESCRIPTION field of the Pull Request when you open it.

        Signed-off-by: Random J Developer <random@developer.example.org>

It would also be helpful if you included an entry in the CHANGELOG.md file describing the change in the PR.  Here is the CHANGELOG.md entry I created for your original PR:

## unreleased
* **Metrics**
  * [CHANGE] The `KubeHpaMaxedOut` alert has (effectively) been renamed `KubeHpaMaxedOutMultiPod`.

I apologize for not catching this earlier and having to ask you to submit a new PR for this.

Regards,
Greg

**Comment by tynsh** [comment]
@gsmith-sas Hi Greg, thanks for the response. Right, I forgot the sign-off. It'll take me until mid next week to get back to this, but I'll send a PR :)

**Comment by gsmith-sas** [comment]
@tynsh  Thanks.  And, as a wise man once said: "Take your time :)".

**Pull request merged**
